solr/core/src/java/org/apache/solr/core/ConfigSetService.java

```diff
@@ -87,20 +78,59 @@ private static ConfigSetService instantiate(CoreContainer coreContainer) {
     }
   }
 
-  private static void bootstrapDefaultConfigSet(ConfigSetService configSetService) throws IOException {
-    if (configSetService.checkConfigExists("_default") == false) {
+  public void bootstrapConfigSet() {
+    // bootstrap _default conf, bootstrap_confdir and bootstrap_conf if provided via system property
+    String confDir = System.getProperty("bootstrap_confdir");
+    boolean boostrapConf = Boolean.getBoolean("bootstrap_conf");
+    try {
+      // _default conf
+      bootstrapDefaultConf();
+      // bootstrap_confdir
+      if (confDir != null) {
+        bootstrapConfDir(confDir);
+      }
+      // bootstrap_conf, in SolrCloud mode
```

Comment:
good point that bootstrap_conf could only make sense in SolrCloud.  So check that we're in SolrCloud mode.  The typical way to do this is check for the existence coreContainer.getZkController
Come to think of it, all of bootstrapConfigSet should be guarded by this.